### ISSUE TITLE
Guide: How to onboard your developers faster with Code Ready Workspaces

### ISSUE THREAD

**Issue by bschrammIBM** [kabanero-io/docs#118]
Need a Guide for `How to onboard your developers faster with an integrated hosted IDE (Eclipse Che) in Kabanero.`
Or perhaps it can be included in the Eclipse Guide see: https://github.com/kabanero-io/docs/issues/102

**Comment by jantley-ibm** [comment]
@SueChaplain please reassign in your team as appropriate

**Comment by SueChaplain** [comment]
We plan to do this work as a separate guide, similar to the work done in #102.

**Comment by SueChaplain** [comment]
@stephenkinder - We could use some technical help on this please. Che is proving difficult to get running. @davehobbs has made more than one attempt but is unable to get a working environment in which to step through the developer flow. Is there someone you can suggest?

**Comment by SueChaplain** [comment]
Some help received from the Codewind team. Che installed and working.

**Comment by marikaj123** [comment]
@SueChaplain - What else do we need such that we can complete the defect.

**Comment by SueChaplain** [comment]
@marikaj123 - We have Che installed and working now, so we're able to start creating the guide to describe how to build an app with the MP collection in Che. It's all new to us, so we might need to reach out for some dev support to get this done for 0.3.0 ..... will keep you up to date.

**Comment by dianak2016** [comment]
@SueChaplain please work with @makandre from Codewind for help.  @gecock has a test scenario for 0.3.0 which depends on this documentation.  Ideally, the doc can be made available before Greg's testing, but at this point the work will likely happen in parallel.

**Comment by makandre** [comment]
The docs for using Codewind on Che is here: https://www.eclipse.org/codewind/installoncloud.html

In particular, to create a Che workspace that has the Codewind components, you would do this:
```
http://<che ingress domain>/f?url=https://raw.githubusercontent.com/eclipse/codewind-che-plugin/master/devfiles/latest/devfile.yaml
```

(Above link is for the latest devfile.  I'll update with the `0.6.0` link once it's available in the next few days)

**Comment by makandre** [comment]
The `0.6.0` link is now available:

```
http://<che ingress domain>/f?url=https://raw.githubusercontent.com/eclipse/codewind-che-plugin/master/devfiles/0.6.0/devfile.yaml
```

**Comment by marikaj123** [comment]
@makandre - is working with the ID team and SVT to complete the work.

**Comment by SueChaplain** [comment]
@marikaj123 / @dianak2016 

Status:

SVT have been attempting an install of Che on OCP 4.2 with help from Andrew. We have taken that dialogue as input to the guide, but this needs much more work to straighten out. I'm not optimistic that this can be ready for tomorrow, but because this is going into the "Guides" area and not "Docs" there is scope to publish this after GA without holding up the release.

Dave is putting an early draft here, which is based on the transcript from this mornings input: https://github.com/kabanero-io/draft-guide-onboarding-with-che

It falls short of the full "onboarding" requirement, but we can modify the title as we extend the guide.

**Comment by kaczyns** [comment]
I am concerned that the draft guide referenced above is not using the Kabanero operator to provision Che.  The operator team was asked to provision the `CheCluster` instance.  You do this by setting `spec.che.enabled: true` in the Kabanero CR instance:
https://kabanero.io/docs/ref/general/kabanero-cr-config.html

If this is not how Che is going to be configured in Kabanero, please let me know so that we can delete all of the code in the operator that is currently provisioning Che.

**Comment by SueChaplain** [comment]
@makandre and @tamdinh - please see the basic draft and provide comments: https://github.com/kabanero-io/draft-guide-onboarding-with-che

**Comment by johnmcollier** [comment]
Echoing the conerns @kaczyns has, the draft guide should be using the Kabanero operator to configure the CheCluster instance (via `spec.che.enabled: true`), and not with Chectl.

**Comment by tam512** [comment]
The steps to enable Che via kabanero CR are documented in my testcase https://github.ibm.com/IBMCloudPak4Apps/icpa-system-test/issues/102#issuecomment-16136685

**Comment by alohr51** [comment]
@SueChaplain @davehobbs Is there anything blocking this guide from being updated with Tim and Johns comments above?

I feel like this is an important guide to get out there since 0.3.0 has been released for awhile now

**Comment by bschrammIBM** [comment]
I think the Che Guide is on hold, per the following email from Alan L. 

"A decision was made yesterday to rremove Che, from the ICPA 4.0.0 release, there is a draft guide that should not go forward.  I should be repurposed, when we pickup CodeReady Workspaces in a followon release.  CodeReady is a RH branded Che.  
 
If there is other doc that talks about the Kabanero branded Che, we need to remove that.

At this point I dont know what we expect will happen with Che in Kabanero.io.  But we will move to CodeReady in ICPA."

**Comment by SueChaplain** [comment]
@gecock @makandre @alohr51 

OK, so we have a draft of the guide available for a technical review. Dave has done the best job he can on this with the information provided. What has held us up is attempting to install/configure a local environment to verify the steps. We felt this was necessary to help orientate the user and pad the instructions with information that helps the workflow. There are still gaps in our knowledge that leave us feeling not 100% happy with the outcome.

The original title of this guide was about onboarding users to Che. What we have provided is a set of install and configure instructions to get Che up and running, with a simple task to initiate a project and run an app to verify that there is working E2E environment.

If someone can provide information about how you then go about onboarding developers to this environment, we can extend the guide, either now before publish, or later.

**Comment by bschrammIBM** [comment]
@ralanlittle stated in an email "I do not think we should be putting out the guide at this point,  a decision on what will be our upstream for CodeReady will come out next week..."

**Comment by SueChaplain** [comment]
@dianak2016 ...... above comment is at odds with your view.

**Comment by dianak2016** [comment]
Alan's comment takes precedence here.  Let's hold then.  Thanks

**Comment by stephenkinder** [comment]
@dianak2016 @SueChaplain So with 0.6.0, we delivered Code Ready Workspaces.  I think this guide is back in play.

**Comment by SueChaplain** [comment]
@davehobbs FYI ..... not sure if you can repurpose what you have or need to start from scratch. 
@makandre .... are you our development contact for Code ready?

**Comment by johnmcollier** [comment]
@SueChaplain I'm the development contact for CodeReady Workspaces

**Comment by SueChaplain** [comment]
FYI we are also talking to the Codewind doc team as they have some work on their backlog and we don't want to duplicate. We'll figure out the assignment between us.

**Comment by SueChaplain** [comment]
Title changed to reflect the guide we think this needs to become: **Guide: How to onboard your developers faster with Code Ready Workspaces**

**Comment by j-c-berger** [comment]
Codewind team is currently working on documentation for Code Ready Workspaces in issue [#2407](https://github.com/eclipse/codewind-docs/pull/487). As of now, no PR for Code Ready documentation is up yet. Our aim is to have a first iteration of Code Ready content by Codewind's next release, which aligns with Kabanero's 0.9.0 release, May 15th.

**Comment by j-c-berger** [comment]
The Codewind team is working on a quick guide on code ready workspaces. Once we complete it for Codewind, we can convert it to be for Kabanero. 

You can see [this Codewind on Che quick lab](https://github.com/beccabau/codewind-che-walkthrough). The Codewind team will adopt this structure and content and refocus it for the code ready workspace quick guide. This quick lab is our starting point for the code ready workspace quick guide.

**Comment by j-c-berger** [comment]
For the Codewind team, I've opened [PR #613-CodeReady Workspace quick guide](https://github.com/eclipse/codewind-docs/pull/613). I can adapt this CRW quick guide for Knowledge Center's purposes. 

This is a first draft, and I'm flexile to take it in any direction that we see is best. Right now, there's a focus on developing in CodeReady Workspaces rather than onboarding developers onto CodeReady Workspaces. 

Note, I took a lot of this content from [Knowledge Center's Installing CodeReady Workspaces](https://www.ibm.com/support/knowledgecenter/SSCSJL_4.1.x/docs/ref/general/installation/installing-codeready-and-codewind.html). I worked on this content a couple months ago and thought that it was applicable here.
